documentation/jetty-documentation/src/main/asciidoc/operations-guide/server/server-logging-server.adoc

```diff
@@ -14,94 +14,86 @@
+You can also configure a default logging level for specific loggers, or arbitrary logger trees:
+* `com.example.MyComponent.LEVEL=DEBUG` (sets logging level of logger `com.example.MyComponent` to `DEBUG`)
+* `com.example.LEVEL=DEBUG` (sets logging level of tree `+com.example.*+` to `DEBUG`)
+
+`org.eclipse.jetty.logging.appender.NAME_CONDENSE=`<boolean>::
 Specifies whether to condense logger names, so that for example `org.eclipse.jetty.util.QueuedThreadPool` becomes `oeju.QueuedThreadPool`.
 Default value is `true`.
 
-org.eclipse.jetty.logging.appender.MESSAGE_ALIGN=<integer>::
+`org.eclipse.jetty.logging.appender.MESSAGE_ALIGN=`<integer>::
 Specifies the column at which the logging `<message>` should be printed.
 The value `0` specifies no alignment.
 Default value is `0`.
 
-org.eclipse.jetty.logging.appender.MESSAGE_ESCAPE=<boolean>::
+`org.eclipse.jetty.logging.appender.MESSAGE_ESCAPE=`<boolean>::
 Specifies whether to escape ISO control characters such as `\r` or `\n` present in the message.
 Character `\r` is replaced with `<` and character `\n` is replaced with `|`; all other ISO control characters are replaced with `?`.
 Default value is `false`.
 
-org.eclipse.jetty.logging.appender.ZONE_ID=<timezone id>::
+`org.eclipse.jetty.logging.appender.ZONE_ID=`<timezone id>::
 Specifies the timezone ID (such as `PST`, or `America/Los_Angeles` or `GMT-8:00`) for the `<datetime>` part of the logging line.
 The empty string specifies the `UTC` timezone.
 Default value is the local timezone.
 
-The `logging-jetty` Jetty module, enabled transitively, provides the configuration file `$JETTY_BASE/resources/jetty-logging.properties` to configure the logging levels, for example:
+`com.example.STACKS=`<boolean>::
+Specifies whether to hide stack traces for some arbitrary logger tree `+com.example.*+`.
+Default value is `true`
 
-----
-$ cd $JETTY_BASE
-$ java -jar $JETTY_HOME/start.jar --add-modules=http
-----
-
-----
-$JETTY_BASE
-├── resources
-│   └── jetty-logging.properties
-└── start.d
-    └── http.ini
-----
-
-.jetty-logging.properties
-[source,properties]
-----
-# Do not condense logger names.
-org.eclipse.jetty.logging.appender.NAME_CONDENSE=false
-
-# By default, log at INFO level all Jetty loggers.
-org.eclipse.jetty.LEVEL=INFO
-
-# However, the Jetty client loggers log at DEBUG level.
-org.eclipse.jetty.client.LEVEL=DEBUG
-----
-
-The logging levels that you can specify in the `jetty-logging.properties` file are the usual SLF4J logging levels, `TRACE`, `DEBUG`, `INFO`, `WARN` and `ERROR`, plus two additional levels:
-
-* `ALL`, which is an alias for `TRACE`
-* `OFF`, which disables entirely the logging (not even `ERROR` level messages are logged)
+`org.eclipse.jetty.logging.jmx.context=`<class name>::
+Configures the JMX context name, for use when xref:og-jmx[debugging via a JMX console].
```

Comment:
I don't really understand what this configuration option does, and Google didn't help me much in figuring out how one might use this feature, so my description here is pretty lackluster.

Reply:
Those options are defined in [StdErrAppender](https://github.com/eclipse/jetty.project/blob/jetty-12.0.x/jetty-core/jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/StdErrAppender.java) which has a lack of javadoc, so hard to say how good your descriptions are.   Is there any magic we can do to put the descriptions in the javadoc and extract them into the documentation?

Some more logging options are configured in [JettyLoggerConfiguration](https://github.com/eclipse/jetty.project/blob/jetty-12.0.x/jetty-core/jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/JettyLoggerConfiguration.java).  It has a thin smear of javadoc, but nothing substantive.

Finally, the jmx config is done by [JettyLoggerFactory](https://github.com/eclipse/jetty.project/blob/jetty-12.0.x/jetty-core/jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/JettyLoggerFactory.java).

Together, these classes cover all the options you have listed, and I can see no more.

@joakime it might be good to give gregp a few paragraphs describing how these 3 classes work together to configure logging... again maybe in the javadoc and/or in the documentation itself.

Reply:
ACK